game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -2568,14 +2570,26 @@ public void validate(final GameData data) throws GameParseException {
   }
 
   private Collection<Territory> getListedTerritories(final String[] list) throws GameParseException {
-    final List<Territory> territories = new ArrayList<>();
+    final Set<Territory> territories = new HashSet<>();
```

Comment:
Was the order always irrelevant or would it make sense to use a LinkedHashSet here?

Reply:
Order is irrelevant as the result of this method was only used to remove elements from another list or used for validation.